### ISSUE TITLE
Inline scripts not working in v5 with space in the filename

### ISSUE THREAD

**Issue by Warzinski** [microsoft/azure-pipelines-tasks#14091]
## Required Information

Entering this information will route you directly to the right team and expedite traction.

**Question, Bug, or Feature?**
*Type*: Bug

**Enter Task Name**: Azure PowerShell

## Environment
- Server - Azure Pipelines

- Agent - Private
    
## Issue Description

Running an inline script does not work on a private agent if the folder path for the agent contains a space.  I've included the logs that show our folder structure.  The script being run is:

Write-Host "Hello, world"

If I change to v4, it runs.  If I use an Azure Pipeline, it runs.  It only appears to be broken in v5.

### Task logs

2020-12-16T16:36:51.6469691Z ========================== Starting Command Output ===========================
2020-12-16T16:36:51.6570374Z ##[debug]Entering Invoke-VstsTool.
2020-12-16T16:36:51.6659691Z ##[debug] Arguments: '-NoLogo -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command ". 'E:\Agents\UPMC Agent Pool\SWAZU2AP6001-1\_work\_temp\59dade0c-9673-4c82-8f7e-48a0aa07ceac.ps1'"'
2020-12-16T16:36:51.6675016Z ##[debug] FileName: 'C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe'
2020-12-16T16:36:51.6689949Z ##[debug] WorkingDirectory: 'E:\Agents\UPMC Agent Pool\SWAZU2AP6001-1\_work\r424\a'
2020-12-16T16:36:51.6734470Z ##[command]"C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe" -NoLogo -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command ". 'E:\Agents\UPMC Agent Pool\SWAZU2AP6001-1\_work\_temp\59dade0c-9673-4c82-8f7e-48a0aa07ceac.ps1'"
2020-12-16T16:36:52.0836613Z . : The term 'E:\Agents\UPMC' is not recognized as the name of a cmdlet, function, script file, or operable program. 
2020-12-16T16:36:52.0840024Z Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
2020-12-16T16:36:52.0841810Z At E:\Agents\UPMC Agent Pool\SWAZU2AP6001-1\_work\_temp\59dade0c-9673-4c82-8f7e-48a0aa07ceac.ps1:3 char:3
2020-12-16T16:36:52.0842700Z + . E:\Agents\UPMC Agent Pool\SWAZU2AP6001-1\_work\_tasks\AzurePowerShe ...
2020-12-16T16:36:52.0843430Z +   ~~~~~~~~~~~~~~
2020-12-16T16:36:52.0844175Z     + CategoryInfo          : ObjectNotFound: (E:\Agents\UPMC:String) [], ParentContainsErrorRecordException
2020-12-16T16:36:52.0845091Z     + FullyQualifiedErrorId : CommandNotFoundException
2020-12-16T16:36:52.0845791Z  
2020-12-16T16:36:52.2158813Z ##[debug]Exit code: 1
2020-12-16T16:36:52.2194149Z ##[debug]Leaving Invoke-VstsTool.
2020-12-16T16:36:52.2485414Z ##[error]PowerShell exited with code '1'.

**Comment by AmrutaKawade** [comment]
@Warzinski can you try putting folder path in double quotes ?

**Comment by Warzinski** [comment]
No, because the folder path isn't a parameter.  That just happens to be where our agents are located.  v4 doesn't have this problem, although I admit that I didn't see a difference in how the folders are treated.

**Comment by AmrutaKawade** [comment]
is it windows / linux agent?

**Comment by Warzinski** [comment]
It's a Windows agent.

Here's the start of the log from when we changed it to v4 and it worked.  You can see the path for the agent is still the same.

2020-12-18T20:40:07.2272435Z ##[debug]VstsTaskSdk 0.11.0 commit 7ff27a3e0bdd6f7b06690ae5f5b63cb84d0f23f4
2020-12-18T20:40:07.4955728Z ##[debug]Entering E:\Agents\UPMC Agent Pool\SWAZU2AP6001-1\_work\_tasks\AzurePowerShell_72a1931b-effb-4d2e-8fd8-f8472a07cb62\4.179.0\azurepowershell.ps1.
2020-12-18T20:40:07.5044680Z ##[debug]Loading resource strings from: E:\Agents\UPMC Agent Pool\SWAZU2AP6001-1\_work\_tasks\AzurePowerShell_72a1931b-effb-4d2e-8fd8-f8472a07cb62\4.179.0\Task.json
2020-12-18T20:40:07.5287817Z ##[debug]Loaded 8 strings.
2020-12-18T20:40:07.5328907Z ##[debug]SYSTEM_CULTURE: 'en-US'
2020-12-18T20:40:07.5349356Z ##[debug]Loading resource strings from: E:\Agents\UPMC Agent Pool\SWAZU2AP6001-1\_work\_tasks\AzurePowerShell_72a1931b-effb-4d2e-8fd8-f8472a07cb62\4.179.0\Strings\resources.resjson\en-US\resources.resjson
2020-12-18T20:40:07.5519098Z ##[debug]Loaded 8 strings.
2020-12-18T20:40:07.5723736Z ##[debug]INPUT_SCRIPTTYPE: 'InlineScript'
2020-12-18T20:40:07.5746348Z ##[debug]INPUT_SCRIPTPATH: 'E:\Agents\UPMC Agent Pool\SWAZU2AP6001-1\_work\r430\a'
2020-12-18T20:40:07.5770801Z ##[debug]INPUT_INLINE: '$port = ([xml](Get-Content ServiceManifest.xml)).ServiceManifest.Resources.Endpoints.Endpoint.Port

**Comment by 20shivangi** [comment]
@Warzinski The issue is occuring on AzurePowerShellV5 in private agent, right?
However the same is working for V4 version in  private agent. Just confirming...

**Comment by Warzinski** [comment]
@20shivangi  That is correct.  I've already provided the logs from v5 where it doesn't work and from v4 where it does work.

**Comment by 20shivangi** [comment]
@Warzinski We are able to reproduce the issue. Will look into it and get back on this.

**Comment by AndrewStrickland** [comment]
Wanted to note this occurs when trying to use ScriptPath as well, not just inline scripts as well.
Version Azure DevOps Server 2020 Update 1

Current agent version: '2.181.2'
```
==============================================================================
Task         : Azure PowerShell
Description  : Run a PowerShell script within an Azure environment
Version      : 5.179.0
Author       : Microsoft Corporation
Help         : https://aka.ms/azurepowershelltroubleshooting
==============================================================================
```
